diff -up plymouth-0.8.2/scripts/plymouth-set-default-theme.in.return-text plymouth-0.8.2/scripts/plymouth-set-default-theme.in
--- plymouth-0.8.2/scripts/plymouth-set-default-theme.in.return-text	2010-04-16 14:40:43.170696779 -0400
+++ plymouth-0.8.2/scripts/plymouth-set-default-theme.in	2010-04-16 14:41:25.856727072 -0400
@@ -66,7 +66,11 @@ function get_default_theme ()
         if [ -z "$THEME_NAME" -o ! -r "${PLYMOUTH_DATADIR}/plymouth/themes/$THEME_NAME/$THEME_NAME.plymouth" ]; then
                 THEME_NAME=$(basename $(readlink ${PLYMOUTH_DATADIR}/plymouth/themes/default.plymouth) .plymouth)
         fi
-        [ -z "$THEME_NAME" ] || echo $THEME_NAME && exit 1
+        if [ -z "$THEME_NAME" ]; then
+                THEME_NAME="text"
+        fi
+
+        echo $THEME_NAME
 }
 
 DO_RESET=0
From 070d2a0a5ad746046ef15065a4fa654cd22906e5 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 14 May 2010 10:56:00 -0400
Subject: [PATCH] [set-default-theme] More fixes to the fallback logic

It still wasn't properly falling back in the event the
configured theme was uninstalled.
---
 scripts/plymouth-set-default-theme.in |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/scripts/plymouth-set-default-theme.in b/scripts/plymouth-set-default-theme.in
index 4631246..105cdd8 100755
--- a/scripts/plymouth-set-default-theme.in
+++ b/scripts/plymouth-set-default-theme.in
@@ -63,10 +63,11 @@ function get_default_theme ()
                 THEME_NAME=$(read_theme_name_from_file ${PLYMOUTH_POLICYDIR}/plymouthd.defaults)
         fi
 
-        if [ -z "$THEME_NAME" -o ! -r "${PLYMOUTH_DATADIR}/plymouth/themes/$THEME_NAME/$THEME_NAME.plymouth" ]; then
+        if [ -z "$THEME_NAME" -o ! -r "${PLYMOUTH_DATADIR}/plymouth/themes/$THEME_NAME/$THEME_NAME.plymouth" \
+             -a -L "${PLYMOUTH_DATADIR}/plymouth/themes/default.plymouth" ]; then
                 THEME_NAME=$(basename $(readlink ${PLYMOUTH_DATADIR}/plymouth/themes/default.plymouth) .plymouth)
         fi
-        if [ -z "$THEME_NAME" ]; then
+        if [ -z "$THEME_NAME" -o ! -r "${PLYMOUTH_DATADIR}/plymouth/themes/$THEME_NAME/$THEME_NAME.plymouth" ]; then
                 THEME_NAME="text"
         fi
 
-- 
1.6.5.2

